EIPS/eip-7901.md

```diff
@@ -0,0 +1,616 @@
+---
+eip: 7901
+title: General Repricing
+description: Gas Cost Repricing to reflect computational complexity and transaction throughput increase
+author: Jacek Glen (@JacekGlen), Lukasz Glen (@lukasz-glen)
+discussions-to: https://ethereum-magicians.org/t/gas-cost-repricing-to-reflect-computational-complexity/23067
+status: Draft
+type: Standards Track
+category: Core
+created: 2025-02-05
+requires: 7883
+---
+
+## Abstract
+
+This proposal revises the gas cost schedule for opcodes, precompiles,
+memory expansion, and data access, prioritizing computational complexity,
+while excluding network-related costs such as state persistence.
+The adjustments aim to enhance gas cost accuracy and rebalance the cost structure.
+
+## Motivation
+
+Gas costs compromise of two components: network (social) cost and computation cost.
+Network cost reflects the blockchain’s effort to maintain state, including adding transactions, storage, logs, calldata, and receipts, while computation cost represents the non-durable processing effort of smart contracts.
+
+While several EIPs (e.g., [EIP-160](eip-160.md), [EIP-1884](eip-1884.md)) have refined network-related gas costs, computational costs have remained largely static since Ethereum’s inception.
+With multiple EVM implementations now stable and optimized, alongside advanced tools, we can assess how well the current gas schedule aligns with hardware workload profiles.
+
+Measurements and estimations depend on various factors,
```

Comment:
reformat for a wider width for a browser to don't add any new lines and let the rendering auto wrap the text